src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -250,7 +250,11 @@ private static Binder GetEnclosingBinder(CSharpSyntaxNode node, int position, Bi
                 {
                     binder = rootBinder.GetBinder(current);
                 }
-                else if (current is ExpressionSyntax && (current.Parent as LambdaExpressionSyntax)?.Body == current)
+                else if (current is ExpressionSyntax && 
```

Comment:
This test is not really about scope designators even though the nodes are also used as the designators.